src/CompositeRheologies.jl

```diff
@@ -794,39 +1005,235 @@ end
             J[j,j] = 1.0
             r[j] = 0.0
         end
     end
 
-    __fill_J_plastic!(static(is_par), j, args)
+    @inline function __fill_J_plastic!(::False, ::True, j, args) #non-parallel, dilatant
+        τ_pl    = x[1]    # if the plastic element is in || with other elements, need to explicitly solve for this
+        P       = x[3]
+
+        args    = merge(args, (τII=τ_pl, P=P))
+        F       = compute_yieldfunction(element,args);  # yield function applied to plastic element
+    
+        ε̇_pl    =  λ̇*∂Q∂τII(element, τ_pl, args)  
+        ε̇vol_pl =  λ̇*∂Q∂P(element, P, args)  
+        
+        r[1]   -=  ε̇_pl                     #  contribution of plastic strainrate to residual
+        r[j+1] +=  ε̇vol_pl                  #  contribution of vol. plastic strainrate to residual
+        if F>=0.0
+            J[1,j] = ∂Q∂τII(element, τ_pl, args)     
+            J[3,j] = -∂Q∂P(element, P, args)      # minus sign because of P sign convention
+            
+            # plasticity is not in a parallel element    
+            J[j,1] = ∂F∂τII(element, τ_pl, args)      # derivative of F vs. τ
+            J[j,j] = ∂F∂λ(element, τ_pl, args)        # derivative of F vs. λ
+            J[j,3] = ∂F∂P(element, P, args)           # derivative of F vs. P
+            r[j] =  -F                          # residual
+        else
+            J[j,j] = 1.0
+            r[j] = 0.0
+        end
+    end
+
+    __fill_J_plastic!(static(is_par), static(is_vol), j, args)
 
     return j
 end
 
+
+
+"""
+    local_iterations_εII(c::CompositeRheology{T,N}, εII_total, εvol_total, args)
+
+This performs nonlinear Newton iterations for `τII` with given `εII_total` for cases where we plastic elements
+"""
+@inline function local_iterations_εvol_εII(
+    c::CompositeRheology{T,N,
+                        Npar,is_par,            # no ||
+                        Nplast, is_plastic,     # with plasticity
+                        Nvol,is_vol,            # with volumetric    
+                        true},                  # with volumetric plasticity        
+    εII_total::_T, 
+    εvol_total::_T, 
+    args; 
+    tol = 1e-6, 
+    verbose = false,
+    τ_initial = nothing, 
+    p_initial = nothing, 
+    ε_init = nothing,
+    max_iter = 1000
+) where {T,N,Npar,is_par, _T, Nplast, is_plastic, Nvol, is_vol}
+   # println("local iterations for εvol_εII")    
+
+    # Compute residual
+    n = 1 + Nplast + Npar + 1;             # total size of unknowns (one added for volumetric plasticity)
+
+    # Initial guess of stress & strainrate
+    if isnothing(τ_initial)
+        τ_initial = compute_τII_harmonic(c, εII_total, args)
+    end
+    if isnothing(p_initial)
+        p_initial = compute_p_harmonic(c, εvol_total, args)
+    end    
+    verbose && println("τII guess = $τ_initial \n  P guess = $p_initial")
+    
+    x    = @MVector zeros(_T, n)
+    x[1] = τ_initial
+
+    j = 1;
+    for i=1:N
+
+        if is_plastic[i] && is_par[i]
```

Comment:
Could you try to use the Static.jl package to replace this branching by multiple dispatch (like [here](https://github.com/JuliaGeodynamics/GeoParams.jl/blob/c93f5f8d2d3152ad96790258e540f725cdf25753/src/CompositeRheologies.jl#L749)) and see if there's a performance gain?